From a2761df4e2c2938b2e9fdef051d5ffc8d10df99f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rapha=C3=ABl=20M=C3=A9lotte?= <raphael.melotte@mind.be>
Date: Tue, 27 Oct 2020 12:38:10 +0100
Subject: [PATCH] hostapd: update wps credentials on reload
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch comes from:
http://lists.infradead.org/pipermail/hostap/2020-August/038719.html

This might not be the best way to update the WPS credentials, see the
following comment:
http://lists.infradead.org/pipermail/hostap/2020-October/039016.html

This however fixes the actual issue with WPS after credentials are
updated, so it's good enough until a more appropriate solution can be
implemented.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>
---
 src/ap/hostapd.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index b37f49f9a..5d8e33911 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -163,6 +163,9 @@ static void hostapd_reload_bss(struct hostapd_data *hapd)
 		wpa_printf(MSG_ERROR, "Could not set SSID for kernel driver");
 		/* try to continue */
 	}
+    /* Update WPS credentials */
+    hostapd_deinit_wps(hapd);
+    hostapd_init_wps(hapd, hapd->conf);
 	wpa_printf(MSG_DEBUG, "Reconfigured interface %s", hapd->conf->iface);
 }
 
-- 
2.28.0

